Sources/WalletConnectKMS/Serialiser/Serializer.swift

```diff
@@ -40,12 +40,13 @@ public class Serializer: Serializing {
     ///   - topic: Topic that is associated with a symetric key for decrypting particular codable object
     ///   - encodedEnvelope: Envelope to deserialize and decrypt
     /// - Returns: Deserialized object
-    public func deserialize<T: Codable>(topic: String, encodedEnvelope: String) throws -> (T, derivedTopic: String?) {
+    public func deserialize<T: Codable>(topic: String, encodedEnvelope: String) throws -> (T, derivedTopic: String?, rawData: String?) {
         let envelope = try Envelope(encodedEnvelope)
         switch envelope.type {
         case .type0:
             let deserialisedType: T = try handleType0Envelope(topic, envelope)
-            return (deserialisedType, nil)
+            let rawRequest = try rawRequest(topic, envelope)
+            return (deserialisedType, nil, rawRequest)
```

Comment:
would make sense that `handleType0Envelope` returns the rawRequest too? like `(T, Data)`
we would do the decryption computation(`codec.decode`) only once then.
now we do it twice: first in `handleType0Envelope` function and then in `try rawRequest(topic, envelope)`

can we also rename `rawRequest`? as it can be a rpc request and response and in theory even more.
just `raw`, `rawDecrypted` `rawDecryptedPayload`?

Reply:
agree with @llbartekll 
if we rename it to rawPayload we could make it not optional, and return it for both type1 and type0 envelopes